### PR TITLE
Events

### PR DESCRIPTION
**Pull request by Chimwemwe-127001**
# Events
> In this feature update:
- [x] Copy the files `calculate.js` and `operate.js` into a `logic/` directory in the project.
- [x] Analyze the files `calculate.js` and `operate.js`
- [x] Import the files in the Calculator component.
- [x] Implement the event handlers.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,33 +1,55 @@
 import React from 'react';
 import '../index.css';
+import calculate from '../logic/calculate';
```
Comment (sinansevgi) [inline src/components/Calculator.js:8]: - [x] For this milestone, we are expecting you to practice what you have learned until now; like passing props to child components, using lifecycle methods, lifting state up. You should use these concepts in your app.

Tip:
_You should divide your `Calculator` component into child components in order to practice these concepts. (For example, you can create a display component for displaying calculations and a keypad component for the keys.)_



From the project requirements ⬇️ 
>Practice what you have learned until now: use state and props, pass props to child components, use lifecycle methods, lift the state up.

**Review by sinansevgi — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
------

Hi @Chimwemwe-127001 , 

Good job so far!
The calculator works like a charm 👏 
There are some issues that you still need to work on to go further but you are almost there!


 ## Required Changes ♻️
_Please check the inline comments for my suggestions_ :arrow_down:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi

------

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (Chimwemwe-127001) [reply src/components/Calculator.js:8]: Hello @sinansevgi. Thanks for the insightful review. I have implemented the required changes as we speak. Hopefully, my project gets approved. :smiley: 
Happy Codding... :computer:

File: src/App.js [hunk 2]
```diff
@@ -1,17 +1,30 @@
+/* eslint-disable react/state-in-constructor */
+/* eslint-disable react/destructuring-assignment */
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
 import Calculator from './components/Calculator';
+import calculate from './logic/calculate';
```
Comment (D4nielJ) [inline src/App.js:6]: [OPTIONAL] When trying to use the modulo with 0, it throws an error. It should behave just the same it does the division by 0. You could try to solve this issue so the Calculator behaves as expected in every case.
image

Comment (D4nielJ) [reply src/App.js:6]: [OPTIONAL] When the `AC` button is pressed, the display of the calculator goes blank. I think the expected behavior, in this case, is for the display to go back to 0. As it is its initial state.

image

Comment (D4nielJ) [reply src/App.js:6]: [OPTIONAL] Your README file has a live demo of another project. Please update this with a link to this project or better remove the link until you have deployed this app. Remember that it is important to keep README's professional as we don't know when a recruiter might visit our projects.

**Review by D4nielJ — APPROVED** [review]
## Hello @Chimwemwe-127001, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 2 above)
Comment (Chimwemwe-127001) [reply src/App.js:6]: Thank you very much @D4nielJ, for your insightful review. I will surely take these lessons above and apply them in my future commits. :smiley: 
Happy coding... :computer: :rocket:

**Pull request merged**